Assignment-10-August-22/userprofile/src/containers/Pages/Dashboard.js

```diff
@@ -1,63 +1,84 @@
-import React, { useState } from "react";
+import React, { useState, useEffect } from "react";
 import { useSelector, useDispatch } from "react-redux";
 import { Card, CardContent, Typography, Box } from "@mui/material";
 import { useNavigate } from "react-router-dom";
 import { Formik, Form } from "formik";
 import * as Yup from "yup";
+import { toast } from "react-toastify";
 
-import { logout } from "../../Store/authSlice";
-import { updateInArray } from "../../Store/userArraySlice";
+import { logout } from "../../store/authSlice";
+import { updateInArray, selectLoggedInUser } from "../../store/userArraySlice";
+import { selectUserArray } from "../../store/userArraySlice";
 
 import ButtonComponent from "../../Components/ButtonComponent";
 import InputField from "../../Components/InputField";
+import setDocumentTitle from "./Title";
+import { ToastContainer } from "react-toastify";
+
 
 export default function Dashboard() {
   const dispatch = useDispatch();
   const navigate = useNavigate();
 
-  const userEmail = useSelector((state) => state.auth.userEmail);
-  const userArray = useSelector((state) => state.userArray.value);
-
-  const loggedInUser = userArray.find((user) => user.email === userEmail);
+  const loggedInUser = useSelector(selectLoggedInUser);
+  const userArray = useSelector(selectUserArray);
 
+  console.log("Logged user", loggedInUser);
   const [editing, setEditing] = useState(false);
 
+  useEffect(() => {
+    setDocumentTitle("Dashboard");
+  }, []);
+
   const initialValues = {
     firstName: loggedInUser?.firstName || "",
     lastName: loggedInUser?.lastName || "",
     country: loggedInUser?.country || "",
+    email: loggedInUser?.email || "",
   };
 
   const validationSchema = Yup.object().shape({
     firstName: Yup.string().required("First Name is required"),
     lastName: Yup.string().required("Last Name is required"),
     country: Yup.string().required("Country is required"),
+    email: Yup.string().email("Please enter a valid email").required(),
   });
 
   const handleLogout = () => {
     dispatch(logout());
     navigate("/login");
   };
 
+  const checkExistence = (userArray,email) =>
+     userArray.some((user) => user.email === email);
+
   const handleUpdate = (values) => {
+    console.log("inside  the update functions");
     if (loggedInUser) {
+
+      if (values.email!==loggedInUser.email  && checkExistence(userArray,values.email))
```

Comment:
we need to use linter properly with our code. 
Add to your IDE or run linter command before commit.